### PR TITLE
Document JSON Schema version updates, and testing

### PR DESCRIPTION
**Pull request by rhiaro**
Documentation for the JSON schema version updates.

Includes a high level overview of the process, and then some working notes of the actual update from draft-04 to 2020-12.

Also started documentation for testing (#8, https://github.com/openownership/data-standard/issues/112) and the metaschema extensions, though there's more to do.

### CODE REVIEW

**Review by kd-ods — COMMENTED** [review]
Amy, these are looking good. I assume that in deleting python.md any info from there that needs to be maintained will go into testing.md? And your working notes in json_schema_...md are you going to keep them for the record, or just delete before we merge this?

File: testing.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Tests for the BODS schema are organised into:
+
+* CSV tests: These validate the codelist CSV files
+* JSON tests: These validate the structure of the schema files, and compliance with the [metaschema](#metaschema)
+* Data tests: These test valid and invalid sample data against the schema files, to check that the schema constrains data as expected
+
+Test files are found in the `data-standard/tests` directory. The tests and a flake8 code quality check are run automatically when a branch is pushed to the `data-standard` repository.
+
+Data for the data tests is organised in several subdirectories under `data-standards/tests/data`.
+
```
Comment (rhiaro) [inline testing.md:28]:
```suggestion

### Code quality

We use  [Black](https://pypi.org/project/black/), * [iSort](https://pypi.org/project/isort/) and [flake8](https://pypi.org/project/flake8/) for code linting.

```

File: json_schema_version_updates.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+* duration
+* uuid (RFC4122)
+
+### 2020-12
+
+* items -> prefixItems
+* additionalItems -> items
+* $dynamicRef/Anchor
+
+Stuff on bundling and compound schema documents
```
Comment (rhiaro) [inline json_schema_version_updates.md:179]:
```suggestion
```

**Comment by rhiaro** [comment]
I retained information from python.md that I thought was useful, and let go of the parts I thought weren't saying a lot. I've now deleted the working notes. Thanks!

**Review by kd-ods — APPROVED** [review]

**Pull request merged**